packages/opencensus-instrumentation-ioredis/test/test-ioredis.ts

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CoreTracer, logger, RootSpan, Span, SpanEventListener, SpanKind} from '@opencensus/core';
+import * as assert from 'assert';
+import * as ioredis from 'ioredis';
+
+import {plugin} from '../src';
+
+/** Collects ended root spans to allow for later analysis. */
+class RootSpanVerifier implements SpanEventListener {
+  endedRootSpans: RootSpan[] = [];
+
+  onStartSpan(span: RootSpan): void {
+    return;
+  }
+  onEndSpan(root: RootSpan) {
+    this.endedRootSpans.push(root);
+  }
+}
+
+/**
+ * Asserts root spans attributes.
+ * @param rootSpanVerifier An instance of rootSpanVerifier to analyse RootSpan
+ *    instances from.
+ * @param expectedName The expected name of the first root span.
+ * @param expectedKind The expected kind of the first root span.
+ */
+function assertSpan(
+    rootSpanVerifier: RootSpanVerifier, expectedName: string,
+    expectedKind: SpanKind, verifyAttribute?: (span: Span) => boolean) {
+  assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 1);
+  assert.strictEqual(rootSpanVerifier.endedRootSpans[0].spans.length, 1);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].name, expectedName);
+  assert.strictEqual(
+      rootSpanVerifier.endedRootSpans[0].spans[0].kind, expectedKind);
+  if (typeof verifyAttribute === 'function') {
+    for (const span of rootSpanVerifier.endedRootSpans[0].spans) {
+      assert(verifyAttribute(span), 'failed to verify attribute');
+    }
+  }
+}
+
+describe('IORedisPlugin', () => {
+  // For these tests, mongo must be runing. Add OPENCENSUS_REDIS_TESTS to run
+  // these tests.
+  const OPENCENSUS_REDIS_TESTS = process.env.OPENCENSUS_REDIS_TESTS;
+  const OPENCENSUS_REDIS_HOST = process.env.OPENCENSUS_REDIS_HOST;
+  const shouldTest = OPENCENSUS_REDIS_TESTS === '1';
+  if (!shouldTest) {
+    console.log('Skipping test-redis. Run REDIS to test');
+  }
+
+  const URL = `redis://${OPENCENSUS_REDIS_HOST || 'localhost'}:6379`;
+  const VERSION = '4.6.2';
+
+  const tracer = new CoreTracer();
+  const rootSpanVerifier = new RootSpanVerifier();
+  let client: ioredis.Redis;
+
+  before((done) => {
+    tracer.start({samplingRate: 1, logger: logger.logger(1)});
+    tracer.registerSpanEventListener(rootSpanVerifier);
+    plugin.enable(ioredis, tracer, VERSION, {}, '');
+    client = new ioredis(URL);
```

Comment:
If redis is not installed locally, do you think this will work?

Reply:
No, it will not work but it's a test file so i would think that it's normal to install it as a dependency to be able to run the test.